### ISSUE TITLE
prefer_inline: `type any` false positive

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#4010]
A `DATA` statement assigning `TYPE ANY` incorrectly produces "prefer inline".

PS: `DATA(lv_var) = iv_param.` is not allowed and correctly produces `DATA definition cannot be generic, AnyType(check_syntax)`.

```abap
REPORT zfoo.

CLASS lcl_test DEFINITION.
  PUBLIC SECTION.
    METHODS test
      IMPORTING
        iv_param TYPE any.
ENDCLASS.

CLASS lcl_test IMPLEMENTATION.
  METHOD test.
    DATA lv_var TYPE string.
    lv_var = iv_param.
  ENDMETHOD.
ENDCLASS.
```

```
/zfoo.prog.abap[12, 10]: Prefer Inline Declarations, LV_VAR(prefer_inline)
```

**Comment by larshp** [comment]
@mbtools this should be fixed, latest version

**Comment by mbtools** [comment]
Putting tokens to a good use 😀 thanks
